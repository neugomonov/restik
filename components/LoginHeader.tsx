import {
	Avatar,
	Box,
	Button,
	Flex,
	HStack,
	Menu,
	MenuButton,
	MenuDivider,
	MenuItem,
	MenuList,
	Stack,
	Text,
	useColorModeValue,
	VStack,
} from "@chakra-ui/react";
import { useSession, signIn, signOut } from "next-auth/react";
import router from "next/router";
import { FiChevronDown } from "react-icons/fi";
export default function LoginHeader() {
	const { data: session } = useSession();
	if (session) {
		return (
			<>
				<Flex alignItems={"center"}>
					<Menu>
						<MenuButton
							py={2}
							transition="all 0.3s"
							_focus={{ boxShadow: "none" }}
						>
							<HStack>
								<Avatar size={"sm"} src={session?.user?.image!} />
								<VStack
									display={{ base: "none", md: "flex" }}
									alignItems="flex-start"
									spacing="1px"
									ml="2"
								>
									<Text fontSize="sm">{session?.user?.name}</Text>
									<Text
										fontSize="xs"
										color={useColorModeValue("gray.600", "gray.300")}
									>
										{session?.user?.email!}
									</Text>
								</VStack>
								<Box display={{ base: "none", md: "flex" }}>
									<FiChevronDown />
								</Box>
							</HStack>
						</MenuButton>
						<MenuList
							bg={useColorModeValue("rgb(255, 255, 255)", "rgb(6, 8, 13)")}
							borderColor={useColorModeValue("gray.200", "gray.700")}
						>
							<MenuItem
								onClick={async () => {
									await router.push("/profile", "/profile", { locale: "ru" });
								}}
							>
								Профиль
							</MenuItem>
							<MenuDivider />
							<MenuItem onClick={() => signOut()}>Выйти</MenuItem>
						</MenuList>
					</Menu>
				</Flex>
			</>
		);
	}
	return (
		<>
			<Stack
				flex={{ base: 1, md: 0 }}
				justify={"center"}
				direction={"row"}
				spacing={6}
			>
				<Button
					variant="outline"
					display={"inline-flex"}
					fontSize={"sm"}
					fontWeight={600}
					colorScheme={"orange"}
					onClick={() => signIn()}
					data-testid="button"
				>
					Войти
				</Button>
			</Stack>
		</>
	);
}
