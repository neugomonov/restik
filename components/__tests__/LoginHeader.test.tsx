import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SessionProvider } from "next-auth/react";
import LoginHeader from "../LoginHeader";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
	render(
		<SessionProvider>
			<LoginHeader />
		</SessionProvider>
	);
});
it("renders button correctly", () => {
	const { getByTestId } = render(
		<SessionProvider>
			<LoginHeader />
		</SessionProvider>
	);
	expect(getByTestId("button")).toHaveTextContent("Войти");
});
it("allows me to log in", () => {
	const { getByTestId } = render(
		<SessionProvider>
			<LoginHeader />
		</SessionProvider>
	);
	userEvent.click(screen.getByTestId("button"));
	expect(getByTestId("button")).toHaveTextContent("Войти");
});
it("matches snapshot", () => {
	const tree = renderer
		.create(
			<SessionProvider>
				<LoginHeader />
			</SessionProvider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
