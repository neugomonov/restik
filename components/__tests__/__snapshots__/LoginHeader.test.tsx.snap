// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-4g6ai3"
  data-projection-id={3}
  draggable={false}
  style={
    {
      "WebkitTouchCallout": "none",
      "WebkitUserSelect": "none",
      "touchAction": "none",
      "userSelect": "none",
    }
  }
>
  <button
    className="chakra-button css-jywq9"
    data-testid="button"
    onClick={[Function]}
    type="button"
  >
    signIn
  </button>
</div>
`;
